src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinValueInstantiator.kt

```diff
@@ -147,30 +104,17 @@ internal class KotlinValueInstantiator(
                 }
             }
 
-            jsonParamValueList[numCallableParameters] = paramVal
-            callableParameters[numCallableParameters] = paramDef
-            numCallableParameters++
+            bucket[idx] = paramVal
         }
 
-        return if (numCallableParameters == jsonParamValueList.size && callable.instanceParameter == null) {
+        // TODO: Is it necessary to call them differently? Direct execution will perform better.
+        return if (bucket.isFullInitialized() && !instantiator.hasInstanceParameter) {
             // we didn't do anything special with default parameters, do a normal call
-            super.createFromObjectWith(ctxt, jsonParamValueList)
+            super.createFromObjectWith(ctxt, bucket.values)
         } else {
-            val accessible = callable.isAccessible
-            if ((!accessible && ctxt.config.isEnabled(MapperFeature.CAN_OVERRIDE_ACCESS_MODIFIERS)) ||
-                (accessible && ctxt.config.isEnabled(MapperFeature.OVERRIDE_PUBLIC_ACCESS_MODIFIERS))
-            ) {
-                callable.isAccessible = true
-            }
-            val callableParametersByName = linkedMapOf<KParameter, Any?>()
-            callableParameters.mapIndexed { idx, paramDef ->
-                if (paramDef != null) {
-                    callableParametersByName[paramDef] = jsonParamValueList[idx]
-                }
-            }
-            callable.callBy(callableParametersByName)
+            instantiator.checkAccessibility(ctxt)
+            instantiator.callBy(bucket)
         }
```

Comment:
Is it necessary to call the contents of the parent class separately?
When I removed this decision and changed it to the following, the test still ran without any problems.

```kotlin
instantiator.checkAccessibility(ctxt)
return instantiator.callBy(bucket)
```

In addition, the benchmark score did not change much due to the faster call to the default no-argument constructor.